.github/workflows/backport_branches.yml

```diff
@@ -40,273 +33,237 @@ jobs:
+            python3 -m json.tool ${{ runner.temp }}/ci_run_data.json
+            echo "::endgroup::"
+
+            {
+              echo 'CI_DATA<<EOF'
+              cat  ${{ runner.temp }}/ci_run_data.json
+              echo 'EOF'
+            } >> "$GITHUB_OUTPUT"
+      - name: Re-create GH statuses for skipped jobs if any
         run: |
-          cd "$GITHUB_WORKSPACE/tests/ci"
-          python3 docker_manifests_merge.py --suffix amd64 --suffix aarch64
-      - name: Upload images files to artifacts
-        uses: actions/upload-artifact@v3
-        with:
-          name: changed_images
-          path: ${{ runner.temp }}/changed_images.json
+            python3 "$GITHUB_WORKSPACE/tests/ci/ci.py" --infile ${{ runner.temp }}/ci_run_data.json --update-gh-statuses
+  BuildDockers:
+    needs: [RunConfig]
+    if: ${{ !failure() && !cancelled() }}
+    uses: ./.github/workflows/reusable_docker.yml
+    with:
+      data: ${{ needs.RunConfig.outputs.data }}
   CompatibilityCheckX86:
-    needs: [BuilderDebRelease]
+    needs: [RunConfig, BuilderDebRelease]
+    if: ${{ !failure() && !cancelled() }}
     uses: ./.github/workflows/reusable_test.yml
     with:
-      test_name: Compatibility check X86
+      test_name: Compatibility check (amd64)
       runner_type: style-checker
+      data: ${{ needs.RunConfig.outputs.data }}
       run_command: |
-        cd "$REPO_COPY/tests/ci"
         python3 compatibility_check.py --check-name "Compatibility check (amd64)" --check-glibc --check-distributions
   CompatibilityCheckAarch64:
-    needs: [BuilderDebAarch64]
+    needs: [RunConfig, BuilderDebAarch64]
+    if: ${{ !failure() && !cancelled() }}
     uses: ./.github/workflows/reusable_test.yml
     with:
-      test_name: Compatibility check X86
+      test_name: Compatibility check (aarch64)
       runner_type: style-checker
+      data: ${{ needs.RunConfig.outputs.data }}
       run_command: |
-        cd "$REPO_COPY/tests/ci"
         python3 compatibility_check.py --check-name "Compatibility check (aarch64)" --check-glibc
 #########################################################################################
 #################################### ORDINARY BUILDS ####################################
 #########################################################################################
   BuilderDebRelease:
-    needs: [DockerHubPush]
+    needs: [RunConfig, BuildDockers]
+    if: ${{ !failure() && !cancelled() }}
     uses: ./.github/workflows/reusable_build.yml
     with:
       build_name: package_release
+      # FIXME: should be 1 everywhere!!!!
       checkout_depth: 0
+      data: ${{ needs.RunConfig.outputs.data }}
   BuilderDebAarch64:
-    needs: [DockerHubPush]
+    needs: [RunConfig, BuilderDebRelease]
+    if: ${{ !failure() && !cancelled() }}
     uses: ./.github/workflows/reusable_build.yml
     with:
       build_name: package_aarch64
+      # FIXME: should be 1 everywhere!!!!
```

Comment:
We regenerate https://github.com/ClickHouse/ClickHouse/blob/master/cmake/autogenerated_versions.txt and https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/System/StorageSystemContributors.generated.cpp

To have it `1 everywhere` it should be preserved in artifacts.

Reply:
comment removed

Reply:
It actually makes sense. My point was to show, what for it's there.